internal/k8s/reconciler/utils_test.go

```diff
@@ -359,3 +362,132 @@ func TestGatewayStatusEqual(t *testing.T) {
+		{name: "different-namespace-with-name-allowed", routeNS: ns1, backendNS: &ns2, backendName: backend2, policyFromNS: ns1, policyToName: &backend2, allowed: true},
+		{name: "mismatched-policy-from-namespace-disallowed", routeNS: ns1, backendNS: &ns2, backendName: backend2, policyFromNS: ns3, policyToName: &backend2, allowed: false},
+		{name: "mismatched-policy-to-name-disallowed", routeNS: ns1, backendNS: &ns2, backendName: backend2, policyFromNS: ns1, policyToName: &backend3, allowed: false},
+	} {
+		// Test each case for both HTTPRoute + TCPRoute which should function identically
+		for _, routeType := range []string{"HTTPRoute", "TCPRoute"} {
+			t.Run(tc.name+"-for-"+routeType, func(t *testing.T) {
+				ctrl := gomock.NewController(t)
+				defer ctrl.Finish()
+				client := mocks.NewMockClient(ctrl)
+
+				group := gw.Group("")
+
+				backendRef := gw.BackendRef{
+					BackendObjectReference: gw.BackendObjectReference{
+						Group: &group,
+						Name:  gw.ObjectName(tc.backendName),
+					},
+				}
+
+				if tc.backendNS != nil {
+					ns := gw.Namespace(*tc.backendNS)
+					backendRef.BackendObjectReference.Namespace = &ns
+				}
+
+				if tc.backendKind != nil {
+					k := gw.Kind(*tc.backendKind)
+					backendRef.Kind = &k
+				}
+
+				var route Route
+				switch routeType {
+				case "HTTPRoute":
+					route = &gw.HTTPRoute{
+						ObjectMeta: meta.ObjectMeta{Namespace: tc.routeNS},
+						TypeMeta:   meta.TypeMeta{APIVersion: "gateway.networking.k8s.io/v1alpha2", Kind: "HTTPRoute"},
+						Spec: gw.HTTPRouteSpec{
+							Rules: []gw.HTTPRouteRule{{
+								BackendRefs: []gw.HTTPBackendRef{{BackendRef: backendRef}},
+							}},
+						},
+					}
+				case "TCPRoute":
+					route = &gw.TCPRoute{
+						ObjectMeta: meta.ObjectMeta{Namespace: tc.routeNS},
+						TypeMeta:   meta.TypeMeta{APIVersion: "gateway.networking.k8s.io/v1alpha2", Kind: "TCPRoute"},
+						Spec: gw.TCPRouteSpec{
+							Rules: []gw.TCPRouteRule{{
+								BackendRefs: []gw.BackendRef{backendRef},
+							}},
+						},
+					}
+				default:
+					require.Fail(t, fmt.Sprintf("unhandled route type %q", routeType))
+				}
+
+				var toName *gw.ObjectName
+				if tc.policyToName != nil {
+					on := gw.ObjectName(*tc.policyToName)
+					toName = &on
+				}
+
+				if tc.backendNS != nil && tc.routeNS != *tc.backendNS {
+					referencePolicy := gw.ReferencePolicy{
+						TypeMeta:   meta.TypeMeta{},
+						ObjectMeta: meta.ObjectMeta{Namespace: *tc.backendNS},
+						Spec: gw.ReferencePolicySpec{
+							From: []gw.ReferencePolicyFrom{{
+								Group:     "gateway.networking.k8s.io",
+								Kind:      gw.Kind(routeType),
+								Namespace: gw.Namespace(tc.policyFromNS),
+							}},
+							To: []gw.ReferencePolicyTo{{
+								Group: "",
+								Kind:  "Service",
+								Name:  toName,
+							}},
+						},
+					}
+
+					throwawayPolicy := gw.ReferencePolicy{
```

Comment:
So, just to make sure, this here just to demonstrate we're filtering out policies via that `validFrom` check, right?

Reply:
Yep yep